doc/computing/randomness.rst

```diff
@@ -0,0 +1,315 @@
+  starts from a different RNG, and the randomly sampled subset of features
+  will be different for each of the 5 folds of the CV procedure.
+
+Here, neither procedure is inherently wrong, and one might prefer one over
+the other depending on the task at hand. It is however important to
+understand how these procedures differ.
+
+.. note::
+    Here, :func:`~sklearn.model_selection.cross_val_score` will use a
+    non-randomized cv splitter (as is the default), so both estimators will
+    be evaluated on the same splits. Also, whether we pass an int or an
+    instance to :func:`~sklearn.datasets.make_classification` isn't relevant
+    for our illustration purpose: what matters is what we pass to the
+    :class:`~sklearn.ensemble.RandomForestClassifier` estimator.
+
+**Cloning**
+
+Another subtle side effect of passing `RandomState` instances is how
+:func:`~sklearn.clone` will work::
+
+    >>> from sklearn import clone
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> import numpy as np
+
+    >>> rng = np.random.RandomState(0)
+    >>> a = RandomForestClassifier(random_state=rng)
+    >>> b = clone(a)
+
+Since a `RandomState` instance was passed to `a`, `a` and `b` are not clones
+in the strict sense, but rather clones in the statistical sense: `a` and `b`
+will still be different models, even after calling `fit(X, y)` on the same
+data. Moreover, `a` and `b` will influence each-other since they share the
+same internal RNG: calling `a.fit` will consume `b`'s RNG, and calling
+`b.fit` will consume `a`'s RNG, since they are the same. This is true for
+any estimators that share a `random_state` parameter; it is not specific to
+clones.
+
+If an int were passed, `a` and `b` would be exact clones and they would not
+influence each other.
+
+This is an important thing to remember because :func:`~sklearn.clone` is
+called everywhere in scikit-learn tools: in particular, most meta-estimators
+that accept non-fitted estimators will in fact call :func:`~sklearn.clone`
+internally (:class:`~sklearn.model_selection.GridSearchCV`,
+:class:`~sklearn.ensemble.StackingClassifier`,
+:class:`~sklearn.calibration.CalibratedClassifierCV`, etc.).
+
+CV splitters
+............
+
+When passed a `RandomState` instance, cv splitters yield different splits
+each time `split` is called. This can lead to dramatic mistakes when
+comparing the performance of different estimators::
+
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.ensemble import GradientBoostingClassifier
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import KFold
+    >>> from sklearn.model_selection import cross_val_score
+    >>> import numpy as np
+
+    >>> rng = np.random.RandomState(0)
+    >>> X, y = make_classification(random_state=rng)
+    >>> rf = RandomForestClassifier(random_state=rng)
+    >>> gbdt = GradientBoostingClassifier(random_state=rng)
```

Comment:
Maybe the most visible way of illustrating the problem would be to use twice the same estimator, and a deterministic one (eg a naive bayes), to make the difference more obvious.